src/sanitize/sanitizeNumbers.ts

```diff
@@ -99,58 +145,67 @@ export function sanitizeNumbers(data: any): any {
 		return jsonSet;
 	}
 
-	if (
-		typeof data === 'number' ||
-		data instanceof AbstractInt ||
-		data instanceof BN
-	) {
-		return data.toString(10);
+	if (data instanceof Map) {
+		return mapTypeSanitizeKeyValue(data);
 	}
 
-	if (data instanceof Object) {
-		if (data._raw != null && data._raw instanceof AbstractInt) {
-			return data._raw.toString(10);
-		}
-
-		if (data.raw != null && data.raw instanceof AbstractInt) {
-			return data.raw.toString(10);
-		}
+	if (data instanceof BN) {
+		return data.toString(10);
+	}
 
-		if (typeof data.toJSON === 'function') {
-			const json = data.toJSON();
+	if (Array.isArray(data)) {
+		return data.map(sanitizeNumbers);
+	}
 
-			if (Array.isArray(json)) {
-				return data.map(sanitizeNumbers);
-			}
+	if (isToJSONable(data)) {
+		// Handle non-codec types that have their own toJSON
+		return sanitizeNumbers(data.toJSON());
+	}
 
-			if (json instanceof Object) {
-				const obj = {};
+	// Pretty much everything non-primitive is an object, so we need to check this last
+	if (isObject(data)) {
+		return Object.entries(data).reduce((sanitizedObject, [key, value]) => {
+			sanitizedObject[key] = sanitizeNumbers(value);
 
-				for (const key of Object.keys(json)) {
-					if (key in data) {
-						// Prefer un-JSON-ified fields
-						obj[key] = sanitizeNumbers(data[key]);
-					} else {
-						obj[key] = sanitizeNumbers(json[key]);
-					}
-				}
+			return sanitizedObject;
+		}, {} as { [prop: string]: AnyJson });
+	}
 
-				return obj;
-			}
+	if (!isAnyJson(data)) {
+		// TODO Should we throw an error here?
+		// Alternatively this could not guarantee AnyJson as a return type and
+		// instead have `unknown` for return type
+		console.error('data could not be forced to `AnyJson` `sanitizeNumber`');
+		console.error(data);
+	}
 
-			return json;
-		}
+	return data as AnyJson;
+}
 
-		if (Array.isArray(data)) {
-			return data.map(sanitizeNumbers);
+/**
+ * Sanitize both the key and values of a map based type, ensuring that the key
+ * is either a number or string.
+ *
+ * @param map Map | CodecMap
+ */
+function mapTypeSanitizeKeyValue(map: Map<unknown, unknown> | CodecMap) {
+	const jsonMap: AnyJson = {};
+
+	map.forEach((value: unknown, key: unknown) => {
+		const nonCodecKey = sanitizeNumbers(key);
+		if (
```

Comment:
I find this `if` statement very difficult to read. Edit: I just realized this is because you are using tabs instead of spaces...